os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -227,33 +227,33 @@ int mm_check_heap_corruption(struct mm_heap_s *heap)
 			mfdbg("=========================================================================================\n");
 			dump_node(heap->mm_heapend[region], "HEAP END NODE");
 			iscorrupt_r = true;
-			end_corrupt = heap->mm_heapend[region] + SIZEOF_MM_ALLOCNODE;
+			end_corrupt = (uint32_t)heap->mm_heapend[region] + SIZEOF_MM_ALLOCNODE;
 			mfdbg("#########################################################################################\n");
 		} else {
 			/* Backward traversal of heap */
 			for (prev = heap->mm_heapend[region], node = (struct mm_allocnode_s *)((char *)prev - MM_PREV_NODE_SIZE(prev));
-					node >= heap->mm_heapstart[region];
+					prev > heap->mm_heapstart[region];
```

Comment:
I think this change belongs to the commit:
os/mm: Fix wrong assertion issue during heap corruption check